### PR TITLE
Switch to Kube 1.14

### PR DESCRIPTION
**Pull request by alexeykazakov**
Requires:
https://github.com/codeready-toolchain/toolchain-common/pull/48 (Merged)
https://github.com/codeready-toolchain/toolchain-common/pull/52 (Merged)
https://github.com/codeready-toolchain/host-operator/pull/99 (Merged)

Paired with https://github.com/codeready-toolchain/toolchain-e2e/pull/32

### CODE REVIEW

**Comment by alexeykazakov** [comment]
https://github.com/codeready-toolchain/host-operator/pull/99 should fix the build failure

File: go.mod [hunk 1]
```diff
@@ -3,57 +3,47 @@ module github.com/codeready-toolchain/host-operator
...
-	k8s.io/apimachinery => k8s.io/apimachinery v0.0.0-20190221213512-86fb29eff628
-	k8s.io/apiserver => k8s.io/apiserver v0.0.0-20190228174905-79427f02047f
-	k8s.io/cli-runtime => k8s.io/cli-runtime v0.0.0-20190228180923-a9e421a79326
-	k8s.io/client-go => k8s.io/client-go v0.0.0-20190228174230-b40b2a5939e4
-	k8s.io/code-generator => k8s.io/code-generator v0.0.0-20181117043124-c2090bec4d9b
-	k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.0.0-20190228175259-3e0149950b0e
-	k8s.io/kube-openapi => k8s.io/kube-openapi v0.0.0-20180711000925-0cf8f7e6ed1d
-	k8s.io/kubernetes => k8s.io/kubernetes v1.14.6
+	github.com/Azure/go-autorest => github.com/Azure/go-autorest v13.0.0+incompatible
+	github.com/codeready-toolchain/toolchain-common => github.com/alexeykazakov/toolchain-common v0.0.0-20191102012635-0f430d8af797
```
Comment (MatousJobanek) [inline go.mod]: Just a reminder to not forget to remove this replace clause ;-)

Comment (alexeykazakov) [reply go.mod]: Removed in https://github.com/codeready-toolchain/host-operator/pull/98/commits/4f97ec68b82232e17b9425cad3d761a8aba95dd3 :)

**Pull request merged**